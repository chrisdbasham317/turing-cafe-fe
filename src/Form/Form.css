form {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

input {
  margin-right: 41px;
  padding: 4px;
  width: 15vw;
}

button {
  background: #008080;
  color: #ffffff;
  width: 15vw;
}