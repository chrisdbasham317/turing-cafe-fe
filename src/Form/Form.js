import React, { Component } from 'react';
import './Form.css';


class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    })
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.addReservation(this.state);
  }

  render() {
    return (
      <form>
        <input type='text' placeholder='Name' name='name' onChange={event => this.handleChange(event)}></input>
        <input type='text' placeholder='Date (mm/dd)' name='date' onChange={event => this.handleChange(event)}></input>
        <input type='text' placeholder='Time' name='time' onChange={event => this.handleChange(event)}></input>
        <input type='number' placeholder='Number of guests' name='number' onChange={event => this.handleChange(event)}></input>
        <button className='button--make-resy' onClick={event => this.handleSubmit(event)}>Make Reservation</button>
      </form>
    )
  }
}

export default Form;