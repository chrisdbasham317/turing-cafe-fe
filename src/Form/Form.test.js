import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper
  let addReservation = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Form
      addReservation={addReservation}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update name state when handleChange is called', () => {
    let mockEvent = {
      target: {
        name: 'name',
        value: 'Chris'
      }
    };
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('name')).toEqual('Chris');
  });

  it('should update name state when handleChange is called', () => {
    let mockEvent = {
      target: {
        name: 'date',
        value: '06/30'
      }
    };
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('date')).toEqual('06/30');
  });
  
  it('should update name state when handleChange is called', () => {
    let mockEvent = {
      target: {
        name: 'time',
        value: '06:30'
      }
    };
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('time')).toEqual('06:30');
  });

  it('should update name state when handleChange is called', () => {
    let mockEvent = {
      target: {
        name: 'number',
        value: 4
      }
    };
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('number')).toEqual(4);
  });
  
  it('should call addReservation when handleSubmit is called', () => {
    const state = {
      name: 'Chris',
      date: '06/30',
      time: '6:30',
      number: 4
    }
    const mockEvent = {
      preventDefault: jest.fn()
    }
    wrapper.setState(state);
    wrapper.instance().handleSubmit(mockEvent)
    expect(addReservation).toHaveBeenCalledWith(state);
  });
})
