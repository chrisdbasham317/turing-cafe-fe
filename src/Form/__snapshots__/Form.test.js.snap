// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="number"
  />
  <button
    className="button--make-resy"
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
